lib/elixir/lib/calendar.ex

```diff
@@ -643,93 +643,7 @@ defmodule NaiveDateTime do
-  NaiveDateTime back to Unix time, you first need to call
-  `DateTime.from_naive/2` with a timezone. For such reason,
-  prefer using `DateTime.from_unix/2` when possible as, opposite
-  to `NaiveDateTime`, it will keep the time zone information.
-
-  ## Examples
-
-      iex> NaiveDateTime.from_unix(1464096368)
-      {:ok, ~N[2016-05-24 13:26:08]}
-
-      iex> NaiveDateTime.from_unix(1432560368868569, :microsecond)
-      {:ok, ~N[2015-05-25 13:26:08.868569]}
-
-  The unit can also be an integer as in `System.time_unit`:
-
-      iex> NaiveDateTime.from_unix(1432560368868569, 1024)
-      {:ok, %NaiveDateTime{calendar: Calendar.ISO, day: 23, hour: 22, microsecond: {211914, 3},
-                           minute: 53, month: 1, second: 43, year: 46302}}
-
-  Negative Unix times are supported, up to -#{@unix_epoch} seconds,
-  which is equivalent to "0000-01-01T00:00:00Z" or 0 gregorian seconds.
-
-      iex> NaiveDateTime.from_unix(-12345678910)
-      {:ok, ~N[1578-10-13 04:44:50]}
-
-  When a Unix time before that moment is passed to `from_unix/2`, `:error` will be returned.
-  """
-  @spec from_unix(integer, :native | System.time_unit) :: {:ok, NaiveDateTime.t}
-  def from_unix(integer, unit \\ :second) when is_integer(integer) do
-    case Calendar.ISO.from_unix(integer, unit) do
-      {:ok, {year, month, day}, {hour, minute, second}, microsecond} ->
-        {:ok, %NaiveDateTime{year: year, month: month, day: day,
-                             hour: hour, minute: minute, second: second, microsecond: microsecond}}
-      {:error, _} = error ->
-        error
-    end
-  end
-
-  @doc """
-  Converts the given Unix time to NaiveDateTime.
-
-  The integer can be given in different unit
-  according to `System.convert_time_unit/3` and it will
-  be converted to microseconds internally.
-
-  Even though Unix times are always in UTC, the time zone
-  is not stored in the naive date time.  Prefer using
-  `DateTime.from_unix/2` when possible as, opposite
-  to `NaiveDateTime`, it will keep the time zone information.
-
-  ## Examples
-
-      iex> NaiveDateTime.from_unix!(1464096368)
-      ~N[2016-05-24 13:26:08]
-
-      iex> NaiveDateTime.from_unix!(1432560368868569, :microsecond)
-      ~N[2015-05-25 13:26:08.868569]
-
-  Negative Unix times are supported, up to -#{@unix_epoch} seconds,
-  which is equivalent to "0000-01-01T00:00:00Z" or 0 gregorian seconds.
-
-      iex> NaiveDateTime.from_unix!(-12345678910)
-      ~N[1578-10-13 04:44:50]
-
-  When a Unix time before that moment is passed to `from_unix/2`, `:error` will be returned.
-  """
-  @spec from_unix!(non_neg_integer, :native | System.time_unit) :: NaiveDateTime.t
-  def from_unix!(integer, unit \\ :second) when is_atom(unit) do
-    case from_unix(integer, unit) do
-      {:ok, datetime} ->
-        datetime
-      {:error, :invalid_unix_time} ->
-        raise ArgumentError, "invalid Unix time #{integer}"
-    end
+    DateTime.utc_now |> DateTime.to_naive
```

Comment:
Because this code can be used in loops (imagine inserting multiple records in the database), I would like to avoid the double conversion. So what if we still call `Calendar.ISO.from_unix` directly? Something like:

```elixir
{:ok, {year, month, day}, {hour, minute, second}, microsecond} =
  Calendar.ISO.from_unix(:os.system_time, :native)
%NaiveDateTime{year: year, month: month, day: day,
               hour: hour, minute: minute, second: second, microsecond: microsecond}